### PR TITLE
fix deleting scenarios

### PR DESCRIPTION
**Pull request by chinook25**

### CODE REVIEW

File: node-backend/scenarioHandler.js [hunk 1]
```diff
@@ -82,47 +83,62 @@ module.exports = function(db) {
...
+    }
+  }
+  
+  function getNewDefaultScenario(scenarioIds, scenarioId) {
+    return _.find(scenarioIds, function(row) {
+      return row.id + '' !== scenarioId;
+    }).id;
+  }
+
+  function changeDefaultScenarioId(workspaceId, scenarioId, callback) {
```
Comment (DanielReid) [inline node-backend/scenarioHandler.js]: again can be inlined

File: node-backend/scenarioHandler.js [hunk 2]
```diff
@@ -82,47 +83,62 @@ module.exports = function(db) {
...
+    WorkspaceRepository.setDefaultScenario(workspaceId, scenarioId, callback);
+  }
+
+  function deleteScenarioAction(scenarioId, setDefaultScenarioResult, callback) {
```
Comment (DanielReid) [inline node-backend/scenarioHandler.js]: I'm not sure, but I *think* that if the `setDefaultScenario` doesn't get called back with an actual value you don't need the redundant argument here and can inline the deleteScenarioAction as well.

File: node-backend/subProblemHandler.js [hunk 3]
```diff
@@ -99,7 +99,7 @@ module.exports = function(db) {
   }
 
   function update(request, response, next) {
-    logger.debug('UPDATE /workspaces/:id/problems/:subProblemId');
+    logger.debug('Update subproblem: '+ request.params.subProblemId);
```
Comment (DanielReid) [inline node-backend/subProblemHandler.js]: the new log line actually gives less information.

File: node-backend/subProblemHandler.js [hunk 4]
```diff
@@ -116,12 +116,12 @@ module.exports = function(db) {
...
     const subproblemId = request.params.subproblemId;
     const workspaceId = request.params.workspaceId;
-    logger.debug('Start deleting subproblem ' + subproblemId);
+    logger.debug('Start deleting subproblem: ' + subproblemId);
     db.runInTransaction(_.partial(deleteTransaction, workspaceId, subproblemId, next), function(error) {
       if (error) {
         util.handleError(error, next);
       } else {
-        logger.debug('done deleting subProblem : ' + JSON.stringify(subproblemId));
+        logger.debug('Done deleting subproblem: ' + JSON.stringify(subproblemId));
```
Comment (DanielReid) [inline node-backend/subProblemHandler.js]: again with the stringify, v. weird. the `+` operator already handles this, surely?

File: test/scenarioHandlerSpec.js [hunk 5]
```diff
@@ -270,28 +277,65 @@ describe('the in scenario handler', () => {
...
+    var setDefaultScenario;
     const scenarioId = 37;
+    const otherScenarioId = 100;
     const subproblemId = 42;
+    const workspaceId = 50;
     const request = {
       params: {
-        id: scenarioId,
-        subproblemId: subproblemId
+        id: scenarioId + '',
```
Comment (DanielReid) [inline test/scenarioHandlerSpec.js]: is this *really* necessary `</impVoice>`

File: node-backend/subProblemHandler.js (hunk 3 above)
Comment (kzalite) [reply node-backend/subProblemHandler.js]: Will print the rest of params too. But I think logger actually doesn't know how to substitute :id etc so less info was printed before the change :)

File: node-backend/scenarioHandler.js (hunk 2 above)
Comment (kzalite) [reply node-backend/scenarioHandler.js]: Would need to see what you mean to act on it.

File: node-backend/subProblemHandler.js (hunk 3 above)
Comment (DanielReid) [reply node-backend/subProblemHandler.js]: yes, but you understood what I meant anyway :P

File: node-backend/scenarioRepository.js [hunk 6]
```diff
@@ -68,12 +68,18 @@ module.exports = function(db) {
...
     const query = 'SELECT id FROM scenario WHERE subproblemId = $1';
     db.query(
       query,
       [subproblemId],
-      callback
+      function(error, result) {
+        if (error) {
+          callback(error);
+        } else {
+          callback(error, result.rows);
```
Comment (DanielReid) [inline node-backend/scenarioRepository.js]: `null` instead of `error` here is a more idiomatic

**Pull request merged**